### PR TITLE
[MGDSTRM-8940] Fix null check when mapping OAuth/SSO trusted certificate

### PR DESCRIPTION
**Pull request by MikeEdgar**
Fix null check when mapping OAuth/SSO trusted certificate

### CODE REVIEW

**Review by vbusch — APPROVED** [review]
lgtm

File: operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java [hunk 1]
```diff
@@ -288,15 +289,14 @@ private Secret ssoClientSecretFrom(ManagedKafka managedKafka, Secret current) {
     }
 
     private Secret ssoTlsSecretFrom(ManagedKafka managedKafka, Secret current) {
-        if (!isKafkaAuthenticationEnabled(managedKafka)) {
```
Comment (shawkins) [inline operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java:291]: Is the isKafkaAuthenticationEnabled check not valid in general?

Comment (MikeEdgar) [reply operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java:291]: It's valid in other contexts (e.g. building the Kafka listeners), but here it's not enough. Not sure if that's what you're asking.

**Review by rareddy — APPROVED** [review]
lgtm

File: operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java [hunk 2]
```diff
@@ -288,15 +289,14 @@ private Secret ssoClientSecretFrom(ManagedKafka managedKafka, Secret current) {
...
-            return null;
-        }
-
-        return buildSecretFrom(ssoTlsSecretName(managedKafka),
-                "Opaque",
-                managedKafka,
-                current,
-                Map.of("keycloak.crt", managedKafka.getSpec().getOauth().getTlsTrustedCertificate()));
+        return Optional.ofNullable(managedKafka.getSpec().getOauth())
+            .map(ManagedKafkaAuthenticationOAuth::getTlsTrustedCertificate)
```
Comment (biswassri) [inline operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java:293]: But if we are changing the structure of this secret, don't we also have to change it in places where we're extracting this the TlsCert/key?

Comment (rareddy) [reply operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java:293]: it is not changing the structure of the secret, it is same. it inlines the isKafkaAuthenticationEnabled call in that first line.

Comment (biswassri) [reply operator/src/main/java/org/bf2/operator/managers/SecuritySecretManager.java:293]: Okay, I understand now how it's the same output. Thanks!

**Pull request merged**